zbus/src/error.rs

```diff
@@ -62,6 +62,8 @@ pub enum Error {
     ///
     /// [MR]: https://dbus.freedesktop.org/doc/dbus-specification.html#message-bus-routing-match-rules
     InvalidMatchRule,
+    /// Invalid `arg0namespace`
+    InvalidArg0Namespace,
```

Comment:
Not sure about this one. We use `InvalidMatchRule` for others so let's use that here as well. Otherwise, this type will keep going bigger and bigger. If we want fine-grained errors, we then need to do that for other errors in MatchBuilder as well.

Also, btw this would be a good candidate for a separate (first) commit.